src/main/scala/org/broadinstitute/dsde/firecloud/utils/StreamingPassthrough.scala

```diff
@@ -0,0 +1,183 @@
+    * currently matched path.
+    *
+    * @param remoteBaseUri the remote system to use as target for passthrough requests
+    */
+  def streamingPassthrough(remoteBaseUri: Uri): Route = {
+    extractMatchedPath { localBasePath =>
+      passthroughImpl(localBasePath, remoteBaseUri)
+    }
+  }
+
+  /**
+    * Passes through, to a remote server, all requests that match or start with the
+    * supplied local path.
+    *
+    * @param passthroughMapping in the form `localBasePath -> remoteBaseUri`, where
+    *                           the localBasePath is the Orchestration-local path
+    *                           which must be matched for passthroughs, and remoteBaseUri
+    *                           is the fully-qualified URL to a remote system
+    *                           to use as target for passthrough requests.
+    */
+  def streamingPassthrough(passthroughMapping: (Uri.Path, Uri)): Route = {
+    passthroughImpl(passthroughMapping._1, passthroughMapping._2)
+  }
+
+  /**
+    * The passthrough implementation:
+    *   - `mapRequest` to transform the incoming request to what we want to send to the remote system
+    *   - `extractRequest` so we have the transformed request as an object
+    *   - call the remote system and reply to the user via `routeResponse` streaming
+   */
+  private def passthroughImpl(localBasePath: Uri.Path, remoteBaseUri: Uri): Route = {
+    mapRequest(transformToPassthroughRequest(localBasePath, remoteBaseUri)) {
+      extractRequest { req =>
+        complete {
+          routeResponse(req)
+        }
+      }
+    }
+  }
+
+
+  /**
+    * Accepts an http request from an end user to Orchestration,
+    * and converts it to a request suitable to send to a remote system
+    * for passthroughs:
+    *   - rewrite the URI to be valid for the remote system
+    *   - strip any http headers, such as Timeout-Access, that are invalid to send to the remote system
+    *
+    * This allows Orchestration to act as an API gateway,
+    * passing the user's request mostly untouched to another service.
+    *
+    * @param req the request inbound to Orchestration
+    * @return the outbound request to be sent to another service
+    */
+  def transformToPassthroughRequest(localBasePath: Uri.Path, remoteBaseUri: Uri)(req: HttpRequest): HttpRequest = {
+    // Convert the URI to the one suitable for the remote system
+    val targetUri = convertToRemoteUri(req.uri, localBasePath, remoteBaseUri)
+
+    // Remove unwanted headers. Akka automatically adds a Timeout-Access to the request
+    // for internal use in managing timeouts; see akka.http.server.request-timeout.
+    // This header is not legal to pass to an external remote system.
+    val targetHeaders = req.headers.filter(_.isNot(`Timeout-Access`.lowercaseName))
+
+    // TODO: what should this log?
+    streamingPassthroughLogger.info(s"${req.method} ${req.uri} => $targetUri")
```

Comment:
I like what we're logging, I'd log a brief explanation for readability in the logs, along these lines:
```suggestion
    streamingPassthroughLogger.info(s"Passthrough API called. Forwarding call: ${req.method} ${req.uri} => $targetUri")
```